esx_service/utils/auth_api.py

```diff
@@ -299,8 +299,20 @@ def _tenant_create(name, description="", vm_list=None, privileges=None):
     if not description:
         description = ""
 
-    logging.debug("_tenant_create: vms=%s", vms)
+    error_info, tenant_list = get_tenant_list_from_db()
```

Comment:
@msterin  I think tenant_create() and adding VMs to it (if specified in options) should be treated as an atomic operation since it was issued in the same command.  If adding VMs to the empty tenant fails, we ll have to cleanup the empty tenant. So creating empty tenant and adding VMs (2 db queries) and then a db commit looks more safe; unless we opt for cleaning the empty tenant in case of VM adding failed. What do you think?